adcc/backends/test_backends_pcm.py

```diff
@@ -0,0 +1,206 @@
+import adcc
+import adcc.backends
+import os
+
+from adcc.misc import expand_test_templates
+from adcc.testdata import static_data
+from adcc.testdata.cache import psi4_data, pyscf_data
+from numpy.testing import assert_allclose
+from adcc.exceptions import InputError
+
+from adcc.adc_pp.environment import block_ph_ph_0_pcm
+from adcc.AdcMatrix import AdcExtraTerm
+
+backends = [b for b in adcc.backends.available()
+            if b in ["psi4", "pyscf"]]
+basissets = ["sto-3g", "cc-pvdz"]
+methods = ["adc1", "adc2", "adc3"]
+
+
+@pytest.mark.skipif(len(backends) == 0, reason="Psi4 and Pyscf backends not found.")
+@expand_test_templates(list(itertools.product(basissets, methods, backends)))
+class TestPCM(unittest.TestCase):
+    def template_pcm_ptlr_formaldehyde(self, basis, method, backend):
+        if method != "adc1":
+            pytest.skip("Data only available for adc1.")
+
+        if backend == "psi4":
+            data = psi4_data
+            run_hf = psi4_run_pcm_hf
+            pcm_options = {"weight": 0.3, "pcm_method": "IEFPCM", "neq": True,
+                           "solvent": "Water"}
+        elif backend == "pyscf":
+            data = pyscf_data
+            run_hf = pyscf_run_pcm_hf
+            pcm_options = {"eps": 78.3553, "eps_opt": 1.78}
+        basename = f"formaldehyde_{basis}_pcm_{method}"
+        result = data[basename]
+
+        scfres = run_hf(static_data.xyz["formaldehyde"], basis, charge=0,
+                        multiplicity=1, conv_tol=1e-12, conv_tol_grad=1e-11,
+                        max_iter=150, pcm_options=pcm_options)
+
+        assert_allclose(scfres.energy_scf, result["energy_scf"], atol=1e-8)
+
+        state = adcc.run_adc(scfres, method=method, n_singlets=5,
+                             conv_tol=1e-7, environment="ptlr")
+
+        # compare ptLR result to LR data
+        assert_allclose(state.excitation_energy,
+                        result["lr_excitation_energy"], atol=5 * 1e-3)
+
+        # Consistency check with values obtained with ADCc
+        assert_allclose(state.excitation_energy,
+                        result["ptlr_adcc_excitation_energy"], atol=1e-6)
+
+        # remove cavity files from PSI4 PCM calculations
+        if backend == "psi4":
+            for cavityfile in os.listdir(os.getcwd()):
+                if cavityfile.startswith(("cavity.off_", "PEDRA.OUT_")):
+                    os.remove(cavityfile)
+
+    def template_pcm_linear_response_formaldehyde(self, basis, method, backend):
+        if method != "adc1":
+            pytest.skip("Reference only exists for adc1.")
+
+        if backend == "psi4":
+            data = psi4_data
+            run_hf = psi4_run_pcm_hf
+            pcm_options = {"weight": 0.3, "pcm_method": "IEFPCM", "neq": True,
+                           "solvent": "Water"}
+        elif backend == "pyscf":
+            data = pyscf_data
+            run_hf = pyscf_run_pcm_hf
+            pcm_options = {"eps": 78.3553, "eps_opt": 1.78}
```

Comment:
This is identical to what you have in the other test above. Can probably be put in a `dict` inside the test class:

```Python
config = {
    "psi4": {"data": psi4_data, "run_hf": psi4_run_pcm_hf, "pcm_options": ...}
    "pyscf": ...
}
```
or sth of the sort...

Reply:
I just add the dict at the end of the file. If i define it as static variable, the program can't find the pcm_hf functions defined below.